app/controllers/users_controller.rb

```diff
@@ -1,10 +1,18 @@
+require 'will_paginate'
+
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :find_user, only: [:show, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :user_role_delete, only: :destroy
+
   def index
-    @users = User.all
+    @users = User.where(activated: true).paginate(page: params[:page], per_page: 20)
```

Comment:
cái này em nên query trong model

Reply:
và sửa dụng [scope trong rails](https://viblo.asia/p/ruby-on-rails-scope-in-rails-E375zgdJKGW). Hãy so sánh scope vs class method khác gì nhau!

Reply:
scope là 1 method bên trong model, luôn trả về 1 active record association. Scope có thể gọi liên tiếp, được định nghĩa bằng cú pháp lambda ngắn gọn, được sử dụng trong các truy vấn đơn giản. Class method được định nghĩa bằng cách sử dụng từ khóa self, được sử dụng cho cho cả truy vấn đơn giản và phức tạp. Khi trả về giá trị nil với class method, có thể gây chết app vì class method trả về exception thay vì 1 ActiveRecord::Relation

Reply:
copy xong rồi áp dụng vào code thử anh xem nào